### ISSUE TITLE
XMLDocument should implement org.w3c.dom.*

### ISSUE THREAD

**Issue by angelozerr** [angelozerr/lsp4xml#172]
To support XPath feature required in 

 * https://github.com/angelozerr/lsp4xml/issues/134
 * https://github.com/angelozerr/lsp4xml/issues/168

XMLDocument , Node, Element, etc must implement  org.w3c.dom.*. I will work on this issue.

**Comment by angelozerr** [comment]
@fbricon @NikolasKomonen now XMLDocument, Element, etc implement org.w3c.dom interfaces and we can benefit with XPath. You can find a test with XPath at https://github.com/angelozerr/lsp4xml/blob/master/org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/dom/XMLDocumentTest.java#L64

Please note this test uses basic XPath. I have not implemented the whole org.w3c.dom interfaces. We should do that step by step. For instance implement previous sibling node to benefit with XPath which requires siblling, etc

In other words we need to add more XPath tests and implements the required org.w3c.dom methods.
